src/main/java/com/damnhandy/uri/template/Expression.java

```diff
@@ -82,6 +85,14 @@
      */
     private Pattern matchPattern;
 
+    private String groupName = uid();
+
+    private static Random RANDOM = new SecureRandom();
```

Comment:
`SecureRandom` is a very expensive way to generate random names, in particular it's blocking, only one thread can generate a name at a time.  If the aim is to ensure that each name this class generates is unique within the scope of this class, then I'd just use an `AtomicLong`, and generate names like `var1`, `var2`, `var3` using `incrementAndGet`. This is fast, non blocking, and guarantees uniqueness up to `Long.MAX_VALUE` names (technically there's no such guarantee with `SecureRandom`).